### ISSUE TITLE
PHP 7.4 nullable properties and TypeHints.TypeHintDeclaration.MissingReturnTypeHint

### ISSUE THREAD

**Issue by rask** [slevomat/coding-standard#731]
When my class has a nullable property:

```php
<?php

class Foo
{
    protected ?Bar $bar = null;

    public function helloWorld() : void
    {
        // noop
    }
}
```

That will result in

```text
--------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------
 7  | ERROR | [x] Method Foo::helloWorld() does not have void return type hint.
    |       |     (SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingReturnTypeHint)
--------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------
```

When I use phpcbf to fix the issue, the result looks like:

```php
<?php

class Foo
{
    protected ?Bar $bar = null;

    public function helloWorld() : void : void
    {
        // noop
    }
}
```

When I change the property to not be nullable (`?Bar -> Bar`), then the sniff works as expected.

EDIT:

I should probably mention that even using `@phpcsSuppress` will not work in this case, at least when I tried.

**Comment by rbairwell** [comment]
Thanks for tracking down what has been causing this and corrupting our code base.

**Comment by rbairwell** [comment]
Possibly related, but given the test code:
```
<?php
declare(strict_types=1);

class Test {

    private Test $abc;

    private ?Test $def;
}
```
(i.e. two properties of the same type - one nullable and one not), I get ```1 | ERROR | [ ] An error occurred during processing; checking has been aborted. The error message was: Undefined index:  in ...\slevomat\coding-standard\SlevomatCodingStandard\Sniffs\ControlStructures\RequireMultiLineTernaryOperatorSniff.php on line 71 (Internal.Exception)```

changing that line to:
```        if (!isset($tokens[$inlineElsePointer]) || $tokens[$inlineThenPointer]['line'] !== $tokens[$inlineElsePointer]['line']) {
```
(i.e. adding the existence check at the start) seems to fix it

**Comment by kukulich** [comment]
There's no support for PHP 7.4 in PHPCS yet.

**Comment by rask** [comment]
@kukulich PHPCS master has support for PHP7.4 typed properties: https://github.com/squizlabs/PHP_CodeSniffer/commit/100a81ca6236dfca27f5d18628ce607dd673c0cc

So I presume this issue will be fixed once the feature linked above becomes available?

**Comment by kukulich** [comment]
Yes, I'll fix it when PHPCS 3.5 is released.
